smk/include/versions.smk

```diff
@@ -0,0 +1,493 @@
+#!/usr/bin/env python
+
+'''
+Print software and database versions for each module
+
+Authors: Judit Szarvas
+'''
+
+import sys
+from os import path
+
+include: 'config_parser.smk'
+
+def get_smk_filename():
+    if '-s' in sys.argv:
+        i = sys.argv.index('-s')
+    elif '--snakefile' in sys.argv:
+        i = sys.argv.index('--snakefile')
+    smk_name = path.splitext(path.basename(sys.argv[i+1]))[0]
+    return smk_name
+
+def get_version_output(smkfilename):
+    smkfiles = ["QC_0", "QC_1", "QC_2", "assembly", "binning_preparation", "mags_generation", "gene_annotation", "gene_abundance", "data_integration", "module_tester"]
+    if smkfilename in smkfiles:
+        result = "{wd}/output/versions/{smk}.flag".format(wd=working_dir, smk=smkfilename)
+        return(result)
+    else:
+        print('WARNING:', smkfilename, 'not set up for printing software versions')
```

Comment:
Should we consider raising an Exception and quitting instead? With large cohorts, the warning gets hidden when Snakemake starts printing its output.

Reply:
It's just a warning, more for future reference to whomever that adds a new smk to the pipeline.